### ISSUE TITLE
Android Auto "For you" system view - how is it populated?

### ISSUE THREAD

**Issue by dxshindeo** [androidx/media#1731]
In Android Auto, on the right side we see a system widget that shows "Now Playing" view.

![image](https://github.com/user-attachments/assets/58151074-dc2f-480d-a917-f4e4df2af1ae)

If we swipe that widget to the right, we get "For you" view (some sort of "Recommended Items" view).

![image](https://github.com/user-attachments/assets/25115f84-2764-403c-a017-6b2d4fe194a0)

Question - how can I control what shows up in this "For you" view?

Ideally I would like to turn it off, because the media items I am playing in my app have to be first acquired dynamically from API and only then playlists can form and be played.

Currently, when a user taps on such media item, I get signal in `ListenableFuture<MediaSession.MediaItemsWithStartPosition> onSetMediaItems` method, which contains said media item in which there is only one usable thing - its id.
At the moment I simply let it die with a silent exception by returning this (so for user nothing happens when he taps the item):

```
return Futures.immediateFuture(
        new MediaSession.MediaItemsWithStartPosition(new ArrayList<>(), 0, 0)
);
```

I would have to initialize data acquisition at this point, but its horrible - there is no saying which data I have to query to API (livestream data, podcast data, etc)... Request could take some time or even fail - there is no loading indicator in this "Recommended Items" view.
So I would like to disable it.

Thanks!

**Comment by OxygenCobalt** [comment]
Also curious about this. My theory is that Android Auto just keeps track of the MediaItems you've published or played and then randomly puts some into that page. I know this because my app's "For You" is just the first N songs in the library, which would also be the first MediaItems you would encounter when opening the app. It's really weird, given that it should be more correct to get the browser root with a suggestions flag.

**Comment by OxygenCobalt** [comment]
That's awesome @NikSatyr.

@marcbaechinger I wonder if it would be best to remap this new package name to setting `LibraryParams.isSuggested` to true.

**Comment by marcbaechinger** [comment]
Sorry, we're not Android Auto and if something isn't documented, then we know as much as you know. Usually you guys know it better than us, as we are not building apps as our daily business.

`com.google.android.googlequicksearchbox` is the search app. I don't think we are going to do something based on the search app accessing the app, simply because we don't know something about it. :)

Can you point us to an official documentation about for you? If think it's understandable we can't do something without it.

**Comment by OxygenCobalt** [comment]
> Can you point us to an official documentation about 'for you;? If think it's understandable we can't do so mething without it.

Sadly, that's the problem @marcbaechinger. This was just randomly added one day and it's not documented anywhere on developer.android.com.

**Comment by dxshindeo** [comment]
@NikSatyr That is an amazing insight, thank you.

For me, the only thing that 100% cleared the cache was restarting the phone.
Then launching the "Desktop Head Unit" I now always get 2 aforementioned "onGetLibraryRoot" calls and can play around with it.

Now - what do you think happened when I returned a MediaItem, which had empty array of children?
**The Android Auto populated "For you" section with another apps content (in this case, Spotify content) :D**

Client once reported this result, but I couldn't evaluate/fix it, because I couldn't replicate it.

So, my options are:
1. return empty children array, but "For you" section will be populated by a completely different apps content;

2. return either default root node or a completely new custom root node (here I am unsure, need to test, whether the items shown will be randomly picked children or children taken in order) and when user clicks on the item in "For you" section and `onSetMediaItems` is called:

2.1. let it silently die by returning this, which results in user clicking on item and nothing happens:
```
return Futures.immediateFuture(
        new MediaSession.MediaItemsWithStartPosition(new ArrayList<>(), 0, 0)
);
```

2.2. get the clicked items "mediaId", which of course we can slightly modify so it contains more info, to become something like `livestream_33` or `podcast_5`, that is, `{content_type}_{item_id}`. 
Then I could look it up in a "source of truth" dataset where my content lives. 
But since that content is only acquired after an API request, the dataset would be empty, if user just launched Android Auto and still hadn't launched my app.
Also, Android Auto does somehow cache those "For you" children and might show cached items, that no longer exist on API side.

--------------------------------

Long story short - I'll have to choose between option 1 and 2.1 since those seem to be the "lesser evil".
Thank you all for pitching in, I'll close this question in a few days.
